lib/topological_inventory/sync/inventory_upload/payload.rb

```diff
@@ -193,6 +193,39 @@ def sources_api_client(tenant = nil)
 
           SourcesApiClient::DefaultApi.new(api_client)
         end
+
+        def topology_inventory_api_client(tenant = nil)
+          api_client = TopologicalInventoryApiClient::ApiClient.new
+
+          if tenant
+            x_rh_identity = Base64.strict_encode64(JSON.dump("identity" => {"account_number" => tenant}))
+            api_client.default_headers.merge!("x-rh-identity" => x_rh_identity)
+          end
+
+          TopologicalInventoryApiClient::DefaultApi.new(api_client)
+        end
+
+        def find_source_in_topology_inventory(source_uid)
+          api_client ||= topology_inventory_api_client(account)
```

Comment:
Better to save variable to api_client call like https://github.com/slemrmartin/topological_inventory-providers-common/blob/master/lib/topological_inventory/providers/common/operations/topology_api_client.rb#L6